headers.go

```diff
@@ -111,6 +119,40 @@ func (h ProtectedHeader) Algorithm() (Algorithm, error) {
 	}
 }
 
+// Critical indicates which protected header labels an application that is
+// processing a message is required to understand.
+//
+// Reference: https://datatracker.ietf.org/doc/html/rfc8152#section-3.1
+func (h ProtectedHeader) Critical() ([]interface{}, error) {
+	value, ok := h[HeaderLabelCritical]
+	if !ok {
+		return nil, nil
+	}
+	criticalLabels, ok := value.([]interface{})
+	if !ok {
+		return nil, errors.New("invalid crit header")
+	}
+	// if present, the array MUST have at least one value in it.
+	if len(criticalLabels) == 0 {
+		return nil, errors.New("empty crit header")
+	}
+	return criticalLabels, nil
+}
+
+// EnsureCritical ensures all critical headers present in the protected bucket.
+func (h ProtectedHeader) EnsureCritical() error {
```

Comment:
Could this function be unexported? It is trivial to implement from outside using `Critical()`